### PR TITLE
3592: Openlist id

### PR DESCRIPTION
**Pull request by BirkAndMe**
https://platform.dandigbib.org/issues/3592

### CODE REVIEW

**Comment by kasperg** [comment]
#370 has a similar issue (although I am personally not that fond of that solution).

File: modules/ding_user/ding_user.module [hunk 1]
```diff
@@ -1101,6 +1101,23 @@ function ding_user_save_creds($auth_res) {
...
+ */
+function ding_user_provider_id($account = NULL) {
+  global $user;
+
+  if ($account === NULL) {
+    $accout = $user;
+  }
+
+  if (ding_provider_implements('user', 'get_provider_id')) {
+    return ding_provider_invoke('user', 'get_provider_id', $account);
```
Comment (kasperg) [inline modules/ding_user/ding_user.module]: :+1:

**Comment by kasperg** [comment]
Please implement it for all providers.

I would prefer to be something that developers have to take into account instead of providing a default provider id value which oddly enough is the same default as is used for the name.

**Comment by BirkAndMe** [comment]
I've implemented the function in the 3 providers, but it simply returns the name in each function, I don't know enough about the 3 providers to figure out an identifier. I can look into it, if there's any documentation for the providers and services I can test up against.

I would assume alma is fine, because it used `name` previously, but I don't know about connie and openruth.

**Comment by kasperg** [comment]
@BirkAndMe Thanks for the update.

I appreciate your comment. That's understandable. The current codebase does use `$creds['name']` for all these providers so while this might not fix the situation for all providers it does provide the proper tools for implementing a fix.

**Comment by ubcudvikler** [comment]
There is an error in [ding_user.module:1111](https://github.com/ding2/ding2/commit/8c38619a544733c393ba10c8e8a8efa02f961ec7#diff-31d3885fdca52f895ca91bc19eb76ebfR1111) where it says "$accout = $user;", it should be "$account = $user;"

**Pull request merged**